drafts/25-xxx-specifications.txt

```diff
@@ -135,6 +135,10 @@ ws09. Whitespace characters are significant in determining token
 ws11. In fixed-form input, macro names are not recognized as such when
       spaces are inserted into their invocations.
 
+ws13. Whitespace must immediately follow the directive tokens
+      'ifdef', 'ifndef', 'elifdef', and 'elifndef' to separate them
```

Comment:
Is this list exhaustive?

Reply:
I forgot about `#undef`.

Other directives can be followed by a broader set of tokens. Whitespace may not be required to separate them, and I don't have a strong opinion about whether whitespace should be required.

We already say that whitespace can be used to separate tokens, but there are places where the whitespace is not needed to separate non-identifier tokens. gfortran and flang accept the following:
```c
#define A+4
#if(1)
#endif
#include"ws003.h"
#warning"This is a warning"
```

Looking at my corpus of Fortran examples, it's not common, but it does appear (3,100 occurrences, mostly with `#if` and `#elif`).